contracts/KyberNetwork.sol

```diff
@@ -286,21 +298,19 @@ contract KyberNetwork {
         }
 
         if( tradeInfo.tradeFailed ) {
-            if( throwOnFailure ) revert();
+            assert( throwOnFailure );
```

Comment:
this will assert only if `throwOnFaile = false`, which is not what we want.
Technically it should be `assert(! throwOnFailure)`. But it is too confusing. So maybe just return to
```
if( throwOnFailure ) revert();
```